sympy/solvers/ode.py

```diff
@@ -4102,12 +4103,13 @@ def _nth_algebraic_match(eq, func):
     constants = iter_numbered_constants(eq)
     constant = lambda: next(constants, None)
 
-    # Like Derivative but "invertible"
+    # A class that behaves like Derivative but is "invertible"
     class diffx(Function):
+        isadiffx = True  # apparently not needed for PY3
         def inverse(self):
-            # We mustn't use integrate here because fx has been replaced by _t
-            # in the equation so integrals will not be correct while solve is
-            # still working.
+            # don't use integrate here because fx has been replaced by _t
+            # in the equation; integrals will not be correct while solve
+            # is at work.
             return lambda expr: Integral(expr, var) + constant()
```

Comment:
I guess it's not trivially possible to move diffx out of the function because it's using closed over variables...

Reply:
That was my experience, yes.